arangod/RestHandler/RestReplicationHandler.cpp

```diff
@@ -2554,16 +2556,29 @@ void RestReplicationHandler::handleCommandHoldReadLockCollection() {
     return;
   }
 
-  if (!body.isObject()) {
+  VPackSlice collection, ttlSlice, idSlice;
+  std::string rebootId;
+
+  if (body.isObject() && body.hasKey("collection") && body.hasKey("ttl") && body.hasKey("id")) {
+    collection = body.get("collection");
+    ttlSlice = body.get("ttl");
+    idSlice = body.get("id");
+  } else {
     generateError(rest::ResponseCode::BAD, TRI_ERROR_HTTP_BAD_PARAMETER,
                   "body needs to be an object with attributes 'collection', "
                   "'ttl' and 'id'");
     return;
   }
 
-  VPackSlice const collection = body.get("collection");
-  VPackSlice const ttlSlice = body.get("ttl");
-  VPackSlice const idSlice = body.get("id");
+  if (body.hasKey("rebootId")) {
```

Comment:
Static strings!
Also there is a helper function to get this via StringRefs and avoid copying the strings
VelocyPackHelper::getStringRef(body, StaticStrings::RebootID, ""); (or getString)

Reply:
very well